test/integration/shoots/operations/clusterautoscaler.go

```diff
@@ -165,6 +166,125 @@ var _ = ginkgo.Describe("Shoot clusterautoscaler testing", func() {
 		framework.ExpectNoError(err)
 	}, testTimeout)
 
+	f.Beta().Serial().CIt("should autoscale a single worker group to/from zero", func(ctx context.Context) {
+		var (
+			shoot = f.Shoot
+
+			origClusterAutoscalerConfig = shoot.Spec.Kubernetes.ClusterAutoscaler.DeepCopy()
+			workerPoolName              = shoot.Spec.Provider.Workers[0].Name
+			testWorkerPoolName          = "ca-test"
+			origWorkers                 = shoot.Spec.Provider.Workers
+		)
+		if shoot.Spec.Provider.Type != "aws" && shoot.Spec.Provider.Type != "azure" {
+			ginkgo.Skip("not applicable")
+		}
+
+		// Create a dedicated worker-pool for cluster autoscaler.
+		testWorkerPool := shoot.Spec.Provider.Workers[0]
+		testWorkerPool.Name = testWorkerPoolName
+		testWorkerPool.Minimum = 0
+		testWorkerPool.Maximum = 2
+		testWorkerPool.Labels = map[string]string{"worker.gardener.cloud/pool": testWorkerPoolName}
+		testTaint := corev1.Taint{
+			Key:    testWorkerPoolName,
+			Effect: corev1.TaintEffectNoSchedule,
+			Value:  testWorkerPoolName,
+		}
+
+		testWorkerPool.Taints = append(testWorkerPool.Taints, testTaint)
```

Comment:
Can you maybe set the test taint directly on the worker pool instead of appending, so you will have only one taint even if the original worker pool might be configured differently in the future?
```suggestion
testWorkerPool.Taints = []corev1.Taint{
			{
				Key:    testWorkerPoolName,
				Effect: corev1.TaintEffectNoSchedule,
				Value:  testWorkerPoolName,
			},
		}
```

Reply:
That's pretty well-thought, thanks for the suggestion.